### DISCUSSION TITLE
How to measure request stages?

### DISCUSSION THREAD

**Discussion by krutoo** [node-fetch/node-fetch#1530] [Q&A]
Hi, how can i measure some request stages like:

- DNS lookup
- time to first byte
- TCP connection established
- TLS Handshake

using `node-fetch`? I want to listen its events for example

**Comment by LinusU** [comment]
Is there any way to do this using `fetch` in the browser? Or a way that is mentioned in the Fetch specification?

**Reply by jimmywarting** [reply]
the best thing we have at mesuring different areas of the things is with [PerformanceResourceTiming](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming) in the browser. node-fetch don't have any support for that cuz there is a lot of things missing
